src/tasks/WasmAppBuilder/PInvokeTableGenerator.cs

```diff
@@ -295,8 +298,15 @@ private string CEntryPoint(PInvokeCallback export)
         }
 
         var method = export.Method;
-        // EntryPoint wasn't specified generate a name for the entry point
-        return _fixupSymbolName($"wasm_native_to_interp_{method.DeclaringType!.Module!.Assembly!.GetName()!.Name!}_{method.DeclaringType.Name}_{method.Name}");
+        string namespaceName = method.DeclaringType?.Namespace ?? string.Empty;
+        string assemblyName = method.DeclaringType?.Module?.Assembly?.GetName()?.Name ?? string.Empty;
+        string declaringTypeName = method.DeclaringType?.Name ?? string.Empty;
+
+        string entryPoint = string.IsNullOrEmpty(namespaceName)
```

Comment:
it might make things a lot simpler to always use the same format whether there's a namespace name or not.

Reply:
Updated.